File: jee/java/src/main/java/fables/kotlin/jee/java/business/KittenBusinessService.java
package fables.kotlin.jee.java.business;

import javax.ejb.Stateless;
import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import java.util.Optional;

/**
 * CRud kitten operations.
 *
 * @author Zeljko Trogrlic
 */
@Stateless
public class KittenBusinessService {

  @PersistenceContext
  protected EntityManager entityManager;

  public Integer add(KittenEntity kitten) {
    entityManager.persist(kitten);
    return kitten.getId();
  }

  public Optional<KittenEntity> find(int id) {
    return Optional.ofNullable(entityManager.find(KittenEntity.class, id));
  }
}

File: jee/converted/src/main/kotlin/fables/kotlin/jee/business/KittenBusinessService.kt
package fables.kotlin.jee.business

import java.util.*
import javax.ejb.Stateless
import javax.persistence.EntityManager
import javax.persistence.PersistenceContext

/**
 * CRud kitten operations.
 *
 * @author Zeljko Trogrlic
 */
@Stateless
open class KittenBusinessService {

    @PersistenceContext
    private lateinit var entityManager: EntityManager

    open fun add(kitten: KittenEntity): Int {
        entityManager.persist(kitten)
        return kitten.id!!
    }

    open fun find(id: Int): Optional<KittenEntity> =
            Optional.ofNullable(entityManager.find(KittenEntity::class.java, id))
}
